cmake/modules/kconfig.cmake

```diff
@@ -157,7 +160,7 @@ set(COMMON_KCONFIG_ENV_SETTINGS
   APP_VERSION_TWEAK_STRING=${APP_VERSION_TWEAK_STRING}
   CONFIG_=${KCONFIG_NAMESPACE}_
   KCONFIG_CONFIG=${DOTCONFIG}
-  BOARD_DIR=${BOARD_DIR}
+  BOARD_DIR=${KCONFIG_BOARD_DIR}
```

Comment:
Seems like a breaking change, depending on how `$(BOARD_DIR)` is used out of tree. I'd suggest adding `KCONFIG_BOARD_DIR` as a separate Kconfig macro.

I suppose that the Kconfig sources should be updated to use `$(KCONFIG_BINARY_DIR)/boards` at some point, but right now, HWMv1 is getting in the way.

Reply:
> Seems like a breaking change

No, for HWMv1 then existing behavior for BOARD_DIR is the same:
https://github.com/zephyrproject-rtos/zephyr/blob/d0dbcc41e1af58a7dae70b6a5d571e2d1bacda38/cmake/modules/kconfig.cmake#L27-L28

For HWMv2 this is not a breaking change because no Zephyr releases until today has included HWMv2.
Upcoming Zephyr v3.7 will be first to include HWMv2, and therefore we can still make changes to how this works for that release.

Reply:
I guess it's fair to point out that the meaning of `BOARD` also changed with HWMv2, but what hasn't changed is the equality of CMake variable and Kconfig macro. The fact that they're different for `BOARD_DIR` might be a head-scratcher.

Reply:
> The fact that they're different for BOARD_DIR might be a head-scratcher.

That's a very valid point.
Will address.